test/views/home_content/challenge_list_test.dart

```diff
@@ -0,0 +1,76 @@
+import "package:flutter/material.dart";
+import "package:flutter_test/flutter_test.dart";
+import "package:proxima/views/home_content/challenge/challenge_card.dart";
+import "package:proxima/views/home_content/challenge/challenge_list.dart";
+
+import "../../mocks/data/challenge_list.dart";
+import "../../mocks/providers/provider_challenge.dart";
+
+void main() {
+  group(
+    "Static testing",
+    () {
+      testWidgets(
+        "Check that there is the correct number of cards, with the correct icons",
+        (tester) async {
+          const data = mockChallengeList;
+
+          await tester.pumpWidget(mockedChallengeListProvider);
+          await tester.pumpAndSettle();
+
+          expect(find.byType(ChallengeList), findsOneWidget);
+          expect(find.byType(ChallengeCard), findsNWidgets(data.length));
+
+          final nGroupChallenge =
+              data.where((challenge) => challenge.isGroupChallenge).length;
+          expect(
+            find.byKey(ChallengeCard.challengeGroupIconKey),
+            findsNWidgets(nGroupChallenge),
+          );
+
+          final nSingleChallenge =
+              data.where((challenge) => !challenge.isGroupChallenge).length;
+          expect(
+            find.byKey(ChallengeCard.challengeSingleIconKey),
+            findsNWidgets(nSingleChallenge),
+          );
+
+          // Challenges can either be single or group
+          expect(nSingleChallenge + nGroupChallenge, data.length);
+        },
+      );
+
+      void expectOneRichText(String text) {
+        // The parameter findRichText from find.text does not appear to work
+        // so I'm forced to use this uglier method.
+        expect(
+          find.byWidgetPredicate(
+            (widget) =>
+                widget is RichText && widget.text.toPlainText().contains(text),
+          ),
+          findsOneWidget,
+        );
+      }
```

Comment:
We could maybe extract this into a "test utils" file as we may want to use it later on

Reply:
Indeed!
Fixed in 3927fe3a441aa92d9e11c8839d929b15c49a6641.